pkg/gateway/server/dispatcher/dispatcher.go

```diff
@@ -224,17 +224,17 @@ func (d *Dispatcher) GetConfiguredAuthProvider(ctx context.Context) (string, err
 // isAuthProviderConfigured checks an auth provider to see if all of its required environment variables are set.
 // Errors are ignored and reported as the auth provider is not configured.
 // Returns: isConfigured (bool)
-func (d *Dispatcher) isAuthProviderConfigured(ctx context.Context, gptscriptClient *gptscript.GPTScript, credCtx []string, toolRef v1.ToolReference) bool {
+func (d *Dispatcher) isAuthProviderConfigured(ctx context.Context, gptscriptClient *gptscript.GPTScript, toolRef v1.ToolReference) bool {
 	if toolRef.Status.Tool == nil {
 		return false
 	}
 
-	cred, err := gptscriptClient.RevealCredential(ctx, credCtx, toolRef.Name)
+	credEnv, err := CredentialEnvForModelProvider(ctx, gptscriptClient, toolRef)
```

Comment:
Why are we calling `CredentialEnvForModelProvider` when we are checking an auth provider? I think we end up using the wrong credential context if we do this.

Reply:
Fixed.